box.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+class Box:
+    def __init__(self, box_dim):
+        self.box_dim = box_dim
+        
+    # wrapping equation
+    def wrap(self, box_dim, coordinates):
+        """Wraps the coordinates within the box dimensions
+
+        Parameters
+        ----------
+        box_dim : np.array
+            Dimensions of the box in reduced units.
+        coordinates : np.array
+            Array of the atomic coordinates.
+
+        Returns
+        -------
+        coordinates : np.array
+            Arrays of the wrapped atomic coordinates.
+        """
+        if len(coordinates.shape) == 1:
+            coordinates -= self.box_dim * np.round(coordinates / self.box_dim)
+        else:
+            coordinates -= self.box_dim * np.round(coordinates / self.box_dim[np.newaxis, :])
+        return coordinates
+        
+    def minimum_image_distance(self,coord_i, coord_j, box_dim):
```

Comment:
I think  you can delete the box_dim argument and in line 45 just call self.box_dim, becuase you already passing self to the method

Reply:
Good eye @jRicciL